mpqp/core/instruction/measurement/basis.py

```diff
@@ -33,27 +35,41 @@ class Basis:
             ``basis_vectors``'s dimensions.
 
     Example:
-        >>> Basis([np.array([1,0]), np.array([0,-1])]).pretty_print()
+        >>> custom_basis = Basis([np.array([1,0]), np.array([0,-1])], symbols=("↑", "↓"))
+        >>> custom_basis.pretty_print()
         Basis: [
             [1, 0],
             [0, -1]
         ]
+        >>> circ = QCircuit([X(0), H(1), CNOT(1, 2), Y(2)])
+        >>> circ.add(BasisMeasure([0, 1, 2], basis=custom_basis, shots=10000))
+        >>> print(run(circ, IBMDevice.AER_SIMULATOR)) # doctest: +SKIP
+        Result: None, IBMDevice, AER_SIMULATOR
+         Counts: [0, 0, 0, 0, 0, 4936, 5064, 0]
+         Probabilities: [0, 0, 0, 0, 0, 0.4936, 0.5064, 0]
+         Samples:
+          State: ↓↑↓, Index: 5, Count: 4936, Probability: 0.4936
+          State: ↓↓↑, Index: 6, Count: 5064, Probability: 0.5064
+         Error: None
 
     """
 
     def __init__(
         self,
         basis_vectors: list[npt.NDArray[np.complex64]],
         nb_qubits: Optional[int] = None,
+        symbols: Optional[tuple[str, str]] = None,
```

Comment:
I think we should also give the possibility to the user to precise the symbols associated with each basis vector, instead of only allowing for pair single qubit symbols. Indeed, not all basis are build using a tensor products of 1-qubit vectors.

So for instance if I have a 2-qubit basis b={u1, u2, u3, u4} that cannot be expressed as {v1v1, v1v2, v2v1, v2v2} we should be able pass all the 2^n symbols for each basis state, so they can appear in the sampling result.

Reply:
makes sense, it would be mutually exclusive with symbols (both cannot be expressed together), something like `basis_vectors_labels` ?

Reply:
Yes indeed 👍

Reply:
done in [b30b039](https://github.com/ColibrITD-SAS/mpqp/pull/101/commits/b30b0394b8b5ce1d925380b7f827a29c5ce9e92a)